tests/test_esm.py

```diff
@@ -49,6 +49,22 @@ def test_enable_esm_install_apt_transport_https_apt_get_options(self):
         self.assertIn(
             'DEBIAN_FRONTEND=noninteractive', self.read_file('apt_get.env'))
 
+    def test_enable_esm_invalid_credentials(self):
+        """If credentials are invalid, an errors is returned."""
+        self.make_fake_binary('curl', command='echo HTTP/1.1 401 Unauthorized')
+        process = self.script('enable-esm', 'user:pass')
+        self.assertEqual(1, process.returncode)
+        self.assertIn('Checking credentials... ERROR', process.stdout)
+        self.assertIn('Invalid credentials', process.stderr)
+
+    def test_enable_esm_error_checking_credentials(self):
+        """If credentials check fails, an error is returned."""
+        self.make_fake_binary('curl', command='exit 1')
```

Comment:
curl won't exit with non-zero if it encounters a 404, for example. You would get the same result in this test if it exited 0. We only check what it prints, right?

Reply:
Correct, I've updated the test (so it also tests the printed HTTP code)